servers/src/main/java/tachyon/worker/block/evictor/EvictorUtils.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block.evictor;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import tachyon.Pair;
+import tachyon.worker.block.BlockMetadataManager;
+import tachyon.worker.block.BlockStoreLocation;
+import tachyon.worker.block.meta.BlockMeta;
+
+public class EvictorUtils {
+  /**
+   * Checks whether the plan is legal for an evictor with cascading eviction feature like
+   * {@link LRUEvictor}. The plan is legal when the requested space can be satisfied, and there is
+   * enough space in lower tier to move in blocks from upper tier.
+   *
+   * @param bytesToBeAvailable requested bytes to be available after eviction
+   * @param plan the eviction plan
+   * @param metaManager the meta data manager
+   * @return true if the above requirements are satisfied, otherwise false.
+   */
+  public static boolean legalCascadingPlan(long bytesToBeAvailable, EvictionPlan plan,
```

Comment:
the assumption here is only blocks in one directory of a tier will be in the plan. But, yes, if the evictor returns a plan as your example, the check will pass, I'll fix it.